libs/vkd3d/command.c

```diff
@@ -2111,6 +2111,66 @@ static void d3d12_command_list_clear_attachment_deferred(struct d3d12_command_li
         attachment->value.depthStencil.stencil = clear_value->depthStencil.stencil;
 }
 
+static void d3d12_command_list_discard_attachment_barrier(struct d3d12_command_list *list,
+        struct d3d12_resource *resource, const VkImageSubresourceLayers *subresource, bool is_bound)
+{
+    const struct vkd3d_vk_device_procs *vk_procs = &list->device->vk_procs;
+    const struct vkd3d_format *format;
+    VkImageMemoryBarrier barrier;
+    VkPipelineStageFlags stages;
+    VkAccessFlags access;
+    VkImageLayout layout;
+
+    if (resource->desc.Flags & D3D12_RESOURCE_FLAG_ALLOW_RENDER_TARGET)
+    {
+        stages = VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT;
+        access = VK_ACCESS_COLOR_ATTACHMENT_WRITE_BIT;
```

Comment:
What about read bits?

Reply:
Do we need those? Reads are undefined after discard.

Reply:
I suppose we can drop the read.